WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.ui.jetpack.restore.usecases
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.FAILED
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.FINISHED
+import org.wordpress.android.fluxc.model.activity.RewindStatusModel.Rewind.Status.RUNNING
+import org.wordpress.android.fluxc.store.ActivityLogStore
+import org.wordpress.android.fluxc.store.ActivityLogStore.FetchRewindStatePayload
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Complete
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Failure.NetworkUnavailable
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Failure.RemoteRequestFailure
+import org.wordpress.android.ui.jetpack.restore.RestoreRequestState.Progress
+import org.wordpress.android.util.NetworkUtilsWrapper
+import javax.inject.Inject
+
+const val DELAY_MILLIS = 10000L
+
+class GetRestoreStatusUseCase @Inject constructor(
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val activityLogStore: ActivityLogStore
+) {
+    suspend fun getRestoreStatus(site: SiteModel, restoreId: Long) = flow {
+        // start off with a delay until "queued" status is implemented in RewindStatusModel
+        delay(DELAY_MILLIS)
```

Comment:
1) Question (❓): Why 10'' seconds?
2) Question (❓): There is another delay at the end of this function, do we need both of them?

Reply:
@ParaskP7 - While testing I discovered there is indeed another state called "queued" which is not accounted for in the existing services. It is not in FluxC either. Once that state is added to fluxc then I can properly handle it in the use case. Looking back at the existing service, they use a "now" parameter to either kick off the status check immediately OR delay for 10 seconds. Since this use case is always run after submittal of the restore request (again mimicking the existing), I just use the delay. I plan on updating FluxC when I update getting progress file names. 

What I will do is move the delay to after the network check - it will have the same effect. in commit 4bd7d42

Reply:
Thanks for the detailed explanation @zwarm !

I am guessing the above is a response to `Question.1`. How about `Question.2`?

Reply:
I removed the second delay in that same commit - moved it so that it can be reused.

Reply:
Oh sorry, missed that, you are right! 🙏